sphinx/config.py

```diff
@@ -163,9 +166,23 @@ def __init__(self, config: Dict[str, Any] = {}, overrides: Dict[str, Any] = {})
 
     @classmethod
     def read(cls, confdir: str, overrides: Dict = None, tags: Tags = None) -> "Config":
-        """Create a Config object from configuration file."""
-        filename = path.join(confdir, CONFIG_FILENAME)
-        namespace = eval_config_file(filename, tags)
+        """Create a Config object from a configuration file."""
+        config_py = path.join(confdir, CONFIG_FILENAME_PY)
+        config_yaml = path.join(confdir, CONFIG_FILENAME_YAML)
+        if path.isfile(config_py):
+            if path.isfile(config_yaml):
+                raise ConfigError(
```

Comment:
I think the initial version should still allow both files, such that one can change to YAML incrementally, and in the end still have dynamic things in ``conf.py`` and static things in ``sphinx.yaml``.

Reply:
No, I disagree.

1. For a version 1 it is best to keep it simple and not add functionality that may need to be later removed
2. We are not removing conf.py, so there is no need for an incremental transition
3. It somewhat defeats the purpose of a static configuration, if you would also have to read `conf.py` to find all the variables

Reply:
and importantly, I think that a decision like this would require more discussion, and isn't strictly a blocker on a first-step implementation, so IMO we should defer that to a future PR after this feature exists and people have had time to use it and get a feel for how it could be improved

Reply:
Well, if you are sitting with an existing project and want the benefit of the static configuration in the YAML file, you will with this be forced to convert _all_ in one go. There is not incremental path to check that you didn't make an error. And if you have just a single dynamic variable (say ``version``) there is no way to use this.
Regarding 3, as argued in the issue: all projects have conventions in one form or another. One such convention would be to ban ``conf.py`` it you want to guarantee static configuration. Why should that block others from using Sphinx in a different way?

Reply:
As I and others have already noted, this is a version 1 implementation that implements the minimum desired functionality and (intentionally) does not preclude the later addition of co-file support.
If you greatly desire such functionality, then I would encourage you to open a follow-up PR to achieve this.
For now this allows "basic" users of sphinx to use the sphinx.yaml, and in no way affects your current sphinx workflow